src/ring_theory/ideal/basic.lean

```diff
@@ -590,6 +590,19 @@ def lift (S : ideal α) (f : α →+* β) (H : ∀ (a : α), a ∈ S → f a = 0
 @[simp] lemma lift_mk (S : ideal α) (f : α →+* β) (H : ∀ (a : α), a ∈ S → f a = 0) :
   lift S f H (mk S a) = f a := rfl
 
+/-- Weaken the relation of a quotient.
+
+This is the `ideal.quotient` version of `quot.factor` -/
+def factor (S T : ideal α) (H : S ≤ T) : S.quotient →+* T.quotient :=
+ideal.quotient.lift S (T^.quotient.mk) (λ x hx, eq_zero_iff_mem.2 (H hx))
+
+@[simp] lemma factor_mk (S T : ideal α) (H : S ≤ T) (x : α) :
+  factor S T H (mk S x) = mk T x := rfl
+
+lemma factor_comp_mk (S T : ideal α) (H : S ≤ T) :
+  ring_hom.comp (factor S T H) (mk S) = mk T :=
+by ext x; rw [ring_hom.comp_apply, quotient.factor_mk]
```

Comment:
Untested:
```suggestion
ring_hom.ext factor_mk
```

Reply:
I switched back to the previous one (this one gave an error)